src/editor/components/Layout/Layout.tsx

```diff
@@ -0,0 +1,53 @@
+import React, {Children, Fragment, PropsWithChildren, ReactElement} from 'react';
+
+import {block} from '../../../utils';
+import {ViewModeItem} from '../../types';
+import ControlPanel from '../ControlPanel/ControlPanel';
+
+import './Layout.scss';
+
+const b = block('editor-layout');
+
+const Left: React.FC<PropsWithChildren> = () => null;
+const Right: React.FC<PropsWithChildren> = () => null;
+
+export interface LayoutProps {
+    mode: ViewModeItem;
+    onModeChange: (mode: ViewModeItem) => void;
+}
+
+const Layout = ({children, mode, onModeChange}: PropsWithChildren<LayoutProps>) => {
+    let left, right;
+
+    function handleChild(child: ReactElement) {
+        switch (child?.type) {
+            case Left:
+                left = child?.props.children;
```

Comment:
won't we have the same potential problem as recently on the blog? when the structure changes and everything falls apart

Reply:
it is just a react pattern, we have it in components like CardBase or in cloud site.  It is like Radio and Radio.Button& Layout relies on its own subcomponents structure only and fetch children from its subcomponents, I think it is okay this way)